pkg/reconciler/taskrun/resources/output_resource.go

```diff
@@ -94,10 +87,12 @@ func AddOutputResources(
 		mkdirSteps := []v1alpha1.Step{v1alpha1.CreateDirStep(images.ShellImage, boundResource.Name, sourcePath)}
 		taskSpec.Steps = append(mkdirSteps, taskSpec.Steps...)
 
-		if allowedOutputResources[resource.GetType()] && taskRun.HasPipelineRunOwnerReference() {
+		needsPvc := false
+		if v1alpha1.AllowedOutputResources[resource.GetType()] && taskRun.HasPipelineRunOwnerReference() {
 			var newSteps []v1alpha1.Step
 			for _, dPath := range boundResource.Paths {
 				newSteps = append(newSteps, as.GetCopyToStorageFromSteps(resource.GetName(), sourcePath, dPath)...)
+				needsPvc = true
```

Comment:
nice, i think this is the key that makes it work which we were missing in our previous attempts @dlorenc ?

b/c in the first attempt we didnt change this at all, and then in #1069 my massive hack was too hacky - i was looking to see if the PVC existed when actually we could just check if this path was bound :woman_facepalming: 

One detail is that we had sort of intended that `path` be something that could be generically provided by users - but im pretty sure that actually didn't work (and hope we'll remove it forever one day soon :pray: ) and this block should only be hit in the PipelineRun case anyway, in which case there's no way for the user to provide the value sooooo

long story short :tada: :tada: :tada: